src/OrleansAzureUtils/Hosting/AzureSilo.cs

```diff
@@ -174,6 +169,8 @@ public bool Start(ClusterConfiguration config, string deploymentId = null, strin
 
             // Bootstrap this Orleans silo instance
 
+            var deploymentId = host.Config.Globals.DeploymentId;
```

Comment:
It is still confusing for the caller, I don't think it is bad to have a breaking change in this case. I think we should allow to pass a config or a deploymentId that will override the value from the config, but not both.